include/vcpkg/base/message-data.inc.h

```diff
@@ -2513,6 +2513,10 @@ DECLARE_MESSAGE(PortVersionMultipleSpecification,
                 (),
                 "",
                 "\"port_version\" cannot be combined with an embedded '#' in the version")
+DECLARE_MESSAGE(PortVersionNonnegativeInteger,
+                (msg::value),
+                "{value} is what the user entered",
+                "port-version (after the '#') must be a non-negative integer, but was {value}")
```

Comment:
1. I argue the placeholder for "user input" *is* `{value}`. That's the placeholder for what happens when we can't say anything about what it looks like because we have failed to make sense of it. I looked into trying to create a separate message arg for it, but this turns out to require a ton of surgery to the entire message system in order to tolerate a description which is not an example to be associated with a message arg, which I don't think is warranted here. I'm not *opposed* in general to a solution which might avoid repeating this sometimes but I don't think it's worth the cost to do such surgery at this time.
2. In my experience without reflecting what the user said back at them in this context, the user often has very little idea what happened. I could see a future where instead of reflecting in the message we show contextual information or file/line number, but that is not what the emitting place of this code can currently see. I also think implicating something like that over this "hey these two fields should use the type we use for them" change would be a bit overkill...